core/src/main/scala/kafka/server/validator/package.scala

```diff
@@ -0,0 +1,27 @@
+package kafka.server
+
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.record.BaseRecords
+import org.apache.kafka.common.record.MemoryRecords
+import org.apache.kafka.common.record.Records
+import org.apache.kafka.common.requests.FetchRequest
+import org.apache.kafka.common.requests.FetchResponse
+import scala.collection.mutable
+
+package object validator {
+  type ErrorElem = (TopicPartition, FetchResponse.PartitionData[Records])
+  type ValidElem = (TopicPartition, FetchRequest.PartitionData)
+
+  type FetchPartitionValidator = RequestValidator[ValidElem, Option[ErrorElem]]
+  type FetchRequestValidator = RequestValidator[(FetchRequest, mutable.Map[TopicPartition, FetchRequest.PartitionData]), Vector[ErrorElem]]
```

Comment:
We have two types of validators for the fetch requests.
1.  FetchRequestValidator which look at the entire request and either accept the all of the fetch partitions or invalidate all of the fetch partitions.
2. FetchPartitionValidator which validate each fetch partition independently.